Feat/setup core models

kingkelly12 opened the pull request.

Commit:
Add models for FoodEntry, Goal, MealPlan, and User with relationships

Commit:
Enhance FoodEntry, Goal, MealPlan, and User models with additional methods for CRUD operations and improve code organization

Commit:
Remove obsolete test files for commands and models

mutuajoseph commented on health_cli/models/foodentry.py:
```diff
@@ -0,0 +1,58 @@
+
+class FoodEntry(Base):
+    __tablename__ = 'food_entries'
+    id = Column(Integer, primary_key=True)
```
add nullable false

mutuajoseph commented on health_cli/models/goal.py:
```diff
@@ -0,0 +1,51 @@
+
+class Goal(Base):
+    __tablename__ = 'goals'
+    id = Column(Integer, primary_key=True)
```
add nullable false

mutuajoseph commented on health_cli/models/mealplan.py:
```diff
@@ -0,0 +1,59 @@
+
+class MealPlan(Base):
+    __tablename__ = 'meal_plans'
+    id = Column(Integer, primary_key=True)
```
add nullable false

mutuajoseph commented on health_cli/models/user.py:
```diff
@@ -0,0 +1,52 @@
+
+class User(Base):
+    __tablename__ = 'users'
+    id = Column(Integer, primary_key=True)
```
add nullable false

mutuajoseph reviewed (changes requested):
Add the db configuration and make sure it works.

Commit:
Add database initialization and session management for SQLAlchemy

Commit:
Add initial empty files for CLI and report model

Commit:
Set 'nullable=False' for primary key columns in Goal, MealPlan, and User models

Commit:
Refactor CLI commands to improve code organization and remove unused functions

Commit:
Enhance goal setting and reporting commands with updates and detailed output

Commit:
Refactor report and meal plan output for clarity and simplifying status messages

Commit:
Implement FoodEntry, Goal, MealPlan, and ReportGenerator controllers with CRUD operations and reporting functionalities

Commit:
Remove FoodEntry, Goal, MealPlan, ReportGenerator controllers and database initialization code

Commit:
added requirement.txt

Commit:
Refactor database structure: remove old database files and implement new configuration for PostgreSQL and SQLite support

Commit:
db update

Commit:
Refactor imports in cli.py and database.py; add User, FoodEntry, and Goal models in models.py

Commit:
Remove redudant code

Commit:
added new test files

Commit:
Add compiled Python bytecode for application and test modules
- Generated __pycache__ files for the main application and test directories.
- Included bytecode for __init__.py, test_cli.py, and test_models.py.
- This change improves the performance of module loading by utilizing precompiled bytecode.

Commit:
Fix app invocation to include program name in cli.py

Commit:
Enhance dashboard command with user feedback and nutrition tracking features

Commit:
Update entry update and delete commands with error handling

Commit:
Add MealPlan model and establish relationship with User

Commit:
Refactor models and database initialization; add MealPlan relationship to User
